packages/frontend/src/components/DashboardFilter/ActiveFilters/index.tsx

```diff
@@ -25,34 +25,38 @@ const ActiveFilters: FC<ActiveFiltersProps> = ({ isEditMode }) => {
 
     return (
         <ActiveDashboardFiltersWrapper>
-            {dashboardFilters.dimensions.map((item, index) => (
-                <ActiveFilter
-                    key={item.id}
-                    isEditMode={isEditMode}
-                    fieldId={item.target.fieldId}
-                    field={fieldMap[item.target.fieldId]}
-                    filterRule={item}
-                    onRemove={() =>
-                        removeDimensionDashboardFilter(index, false)
-                    }
-                    onUpdate={(value) =>
-                        updateDimensionDashboardFilter(value, index, false)
-                    }
-                />
-            ))}
-            {dashboardTemporaryFilters.dimensions.map((item, index) => (
-                <ActiveFilter
-                    key={item.id}
-                    isEditMode={isEditMode}
-                    fieldId={item.target.fieldId}
-                    field={fieldMap[item.target.fieldId]}
-                    filterRule={item}
-                    onRemove={() => removeDimensionDashboardFilter(index, true)}
-                    onUpdate={(value) =>
-                        updateDimensionDashboardFilter(value, index, true)
-                    }
-                />
-            ))}
+            {dashboardFilters.dimensions
+                .filter((item) => !!fieldMap[item.target.fieldId])
```

Comment:
Since you are just doing a filter here, it doesn't get removed in the DB, right? So does the filter come back if you add back a chart that *was* using it?

Will that be surprising to people? Or maybe that is actually what we want.

Reply:
Cool! I'm glad the video explained it 😅